packages/components/message/src/message-method.ts

```diff
@@ -6,116 +6,121 @@ import { messageConfig } from '@element-plus/components/config-provider/src/conf
-import type { ComponentPublicInstance, VNode } from 'vue'
 
 const instances: MessageQueue = []
 let seed = 1
 
 // TODO: Since Notify.ts is basically the same like this file. So we could do some encapsulation against them to reduce code duplication.
 
-const message: MessageFn & Partial<Message> = function (options = {}) {
-  if (!isClient) return { close: () => undefined }
-  if (isNumber(messageConfig.max) && instances.length >= messageConfig.max) {
-    return { close: () => undefined }
-  }
+const message: MessageFn & Partial<Message> & { _context: AppContext | null } =
+  function (options = {}, context?: AppContext | null) {
+    if (!isClient) return { close: () => undefined }
+    if (isNumber(messageConfig.max) && instances.length >= messageConfig.max) {
+      return { close: () => undefined }
+    }
 
-  if (
-    !isVNode(options) &&
-    typeof options === 'object' &&
-    options.grouping &&
-    !isVNode(options.message) &&
-    instances.length
-  ) {
-    const tempVm: any = instances.find(
-      (item) =>
-        `${item.vm.props?.message ?? ''}` ===
-        `${(options as any).message ?? ''}`
-    )
-    if (tempVm) {
-      tempVm.vm.component!.props.repeatNum += 1
-      tempVm.vm.component!.props.type = options?.type
-      return {
-        close: () =>
-          ((
-            vm.component!.proxy as ComponentPublicInstance<{ visible: boolean }>
-          ).visible = false),
+    if (
+      !isVNode(options) &&
+      typeof options === 'object' &&
+      options.grouping &&
+      !isVNode(options.message) &&
+      instances.length
+    ) {
+      const tempVm: any = instances.find(
+        (item) =>
+          `${item.vm.props?.message ?? ''}` ===
+          `${(options as any).message ?? ''}`
+      )
+      if (tempVm) {
+        tempVm.vm.component!.props.repeatNum += 1
+        tempVm.vm.component!.props.type = options?.type
+        return {
+          close: () =>
+            ((
+              vm.component!.proxy as ComponentPublicInstance<{
+                visible: boolean
+              }>
+            ).visible = false),
+        }
       }
     }
-  }
 
-  if (typeof options === 'string' || isVNode(options)) {
-    options = { message: options }
-  }
+    if (typeof options === 'string' || isVNode(options)) {
+      options = { message: options }
+    }
 
-  let verticalOffset = options.offset || 20
-  instances.forEach(({ vm }) => {
-    verticalOffset += (vm.el?.offsetHeight || 0) + 16
-  })
-  verticalOffset += 16
-
-  const { nextZIndex } = useZIndex()
-
-  const id = `message_${seed++}`
-  const userOnClose = options.onClose
-  const props: Partial<MessageProps> = {
-    zIndex: nextZIndex(),
-    offset: verticalOffset,
-    ...options,
-    id,
-    onClose: () => {
-      close(id, userOnClose)
-    },
-  }
+    let verticalOffset = options.offset || 20
+    instances.forEach(({ vm }) => {
+      verticalOffset += (vm.el?.offsetHeight || 0) + 16
```

Comment:
vm?.el

Reply:
Nothing changed here, this is just formatting.